hibernate-core/src/main/java/org/hibernate/metamodel/internal/AttributeFactory.java

```diff
@@ -213,7 +213,7 @@ public Property getPropertyMapping() {
 			case EMBEDDABLE: {
 				final Component component = (Component) typeContext.getValue();
 				final EmbeddableTypeImpl<Y> embeddableType = new EmbeddableTypeImpl<Y>(
-						typeContext.getBindableType(),
+						component.getComponentClass(),
```

Comment:
with dynamic map components the value of `componentClassName` is null so `component.getComponentClass()` throws an Exception. 
I think the code should be changed to
```
Class javaType = null;
if ( component.getComponentClassName() == null ) {
      javaType = typeContext.getBindableType();
}
else {
      javaType = component.getComponentClass();
}
final EmbeddableTypeImpl<Y> embeddableType = new EmbeddableTypeImpl<Y>(
	javaType,
	typeContext.getAttributeMetadata().getOwnerType(),
	(ComponentType) typeContext.getValue().getType()
);
```